src/main/webapp/resources/js/pages/projects/samples-metadata-import/components/SampleMetadataImportSelectFile.tsx

```diff
@@ -25,61 +25,102 @@ export function SampleMetadataImportSelectFile(): JSX.Element {
-    beforeUpload: (file: RcFile) => {
+    customRequest: () => {
+      navigate(`/${projectId}/sample-metadata/upload/columns`);
+    },
+    beforeUpload: async (file: RcFile) => {
       try {
-        setLoading(true);
-        const reader = new FileReader();
-        if (reader.readAsBinaryString) {
-          reader.onload = async () => {
-            const workbook: WorkBook = XLSX.read(reader.result, {
-              type: "binary",
-              raw: true,
-            });
-            const { SheetNames } = workbook;
-            const [firstSheet] = SheetNames;
-            const rows: MetadataItem[] = XLSX.utils.sheet_to_json(
-              workbook.Sheets[firstSheet],
-              {
-                rawNumbers: false,
-              }
-            );
-            const cleanRows = JSON.parse(
-              JSON.stringify(rows).replace(/"\s+|\s+"/g, '"')
-            );
-            await dispatch(setHeaders({ headers: Object.keys(cleanRows[0]) }));
-            await dispatch(setMetadata(cleanRows));
-          };
-          reader.readAsBinaryString(file);
+        const data = await readFileContents(file);
+        const workbook: XLSX.WorkBook = XLSX.read(data, {
+          type: "binary",
+          raw: true,
+        });
+        const { SheetNames } = workbook;
+        const [firstSheet] = SheetNames;
+        // Not loving how I'm parsing the file twice to get the raw headers
+        // See https://docs.sheetjs.com/docs/api/utilities
+        const headers: string[] = XLSX.utils.sheet_to_json(
+          workbook.Sheets[firstSheet],
+          {
+            header: 1,
+          }
+        )[0];
+        const duplicateHeaders = headers.filter(
+          (e, i, a) => a.indexOf(e) !== i
+        );
+        if (duplicateHeaders.length > 0) {
+          setLoading(false);
+          setValid(false);
+          return false;
+        } else {
+          setValid(true);
+          const rows: MetadataItem[] = XLSX.utils.sheet_to_json(
+            workbook.Sheets[firstSheet],
+            {
+              rawNumbers: false,
+            }
+          );
+          const cleanRows = JSON.parse(
+            JSON.stringify(rows).replace(/"\s+|\s+"/g, '"')
+          );
+          await dispatch(setHeaders({ headers: Object.keys(cleanRows[0]) }));
+          await dispatch(setMetadata(cleanRows));
           notification.success({
             message: i18n("SampleMetadataImportSelectFile.success", file.name),
           });
-          navigate(`/${projectId}/sample-metadata/upload/columns`);
         }
       } catch (error) {
         setLoading(false);
         setStatus("error");
         notification.error({
           message: i18n("SampleMetadataImportSelectFile.error", file.name),
         });
+        return false;
       }
-      return false;
+      return true;
     },
   };
 
   return (
     <SampleMetadataImportWizard current={0} status={status}>
       <Spin spinning={loading}>
+        {!valid && (
```

Comment:
Add spacing between the `ErrorAlert` and the `DragUpload`. Probably use `marginBottom: SPACE_XS` on the `ErrorAlert`

Reply:
Please see 3dbcfcc176217a7368307fb3ea8bd4d351ec2105.